### PR TITLE
Multiple dependency bumps

### PR DESCRIPTION
**Pull request by rnveach**
Identified after noticing https://github.com/checkstyle/checkstyle/pull/12707#discussion_r1094701977 and running `mvn versions:display-dependency-updates`.

4-5 dependencies are out of date.

Based on below evidence, it looks like we stopped getting some updates around 10/25/2022 - 10/29/2022 .

`versions-maven-plugin` got an update around the same time.
https://mvnrepository.com/artifact/org.codehaus.mojo/versions-maven-plugin
2.13.0 release on 10/23.
It's latest release is 2.14.2 which was released on 12/24.

@romani ping

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -210,7 +210,7 @@
     <maven.site.plugin.version>3.12.1</maven.site.plugin.version>
     <maven.spotbugs.plugin.version>4.7.2.1</maven.spotbugs.plugin.version>
     <maven.pmd.plugin.version>3.20.0</maven.pmd.plugin.version>
-    <pmd.version>6.50.0</pmd.version>
+    <pmd.version>6.54.0</pmd.version>
```
Comment (rnveach) [inline pom.xml:213]: https://github.com/pmd/pmd/releases
6.54 was released on 1/28.
6.51 was released on 10/29.
6.50 was released on 9/30.

File: pom.xml [hunk 2]
```diff
@@ -227,10 +227,10 @@
     <pitest.plugin.output.formats>HTML,XML</pitest.plugin.output.formats>
     <pitest.plugin.timeout.constant>50000</pitest.plugin.timeout.constant>
     <pitest.plugin.threads>4</pitest.plugin.threads>
-    <pitest.junit5.plugin.version>1.1.0</pitest.junit5.plugin.version>
+    <pitest.junit5.plugin.version>1.1.2</pitest.junit5.plugin.version>
     <pitest.accelerator.junit5.plugin.version>1.0.4</pitest.accelerator.junit5.plugin.version>
     <sonar.test.exclusions>**/test/resources/**/*,**/it/resources/**/*</sonar.test.exclusions>
-    <junit.version>5.9.1</junit.version>
+    <junit.version>5.9.2</junit.version>
```
Comment (rnveach) [inline pom.xml:233]: https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-api
5.9.2 was released on 1/10.
5.9.1 was released on 9/20.

File: pom.xml [hunk 3]
```diff
@@ -227,10 +227,10 @@
     <pitest.plugin.output.formats>HTML,XML</pitest.plugin.output.formats>
     <pitest.plugin.timeout.constant>50000</pitest.plugin.timeout.constant>
     <pitest.plugin.threads>4</pitest.plugin.threads>
-    <pitest.junit5.plugin.version>1.1.0</pitest.junit5.plugin.version>
+    <pitest.junit5.plugin.version>1.1.2</pitest.junit5.plugin.version>
```
Comment (rnveach) [inline pom.xml:230]: https://mvnrepository.com/artifact/org.pitest/pitest-junit5-plugin
1.1.2 was released on 1/12.
1.1.1 was released on 1/4.
1.1.0 was released on 9/26.

File: pom.xml [hunk 4]
```diff
@@ -298,13 +298,13 @@
     <dependency>
       <groupId>org.junit-pioneer</groupId>
       <artifactId>junit-pioneer</artifactId>
-      <version>1.7.1</version>
+      <version>2.0.0</version>
```
Comment (rnveach) [inline pom.xml:301]: https://mvnrepository.com/artifact/org.junit-pioneer/junit-pioneer
2.0.0 was released on 12/21.
1.7.2 was released on 11/9.
1.7.1 was released on 5/25.

I am confused as I was told the new version was 2.0.0 but maven says it is actually `2.0.0-RC1`.

File: pom.xml [hunk 5]
```diff
@@ -298,13 +298,13 @@
...
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>com.tngtech.archunit</groupId>
       <artifactId>archunit-junit5</artifactId>
-      <version>1.0.0</version>
+      <version>1.0.1</version>
```
Comment (rnveach) [inline pom.xml:307]: https://mvnrepository.com/artifact/com.tngtech.archunit/archunit-junit5
1.0.1 was released on 11/21.
1.0.0 was released on 10/3.

File: pom.xml (hunk 1 above)
Comment (rnveach) [reply pom.xml:213]: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/17201/workflows/932a838c-765b-4619-85fb-64b47b8927f8/jobs/240411?invite=true#step-103-96
> You have 26 PMD violations.

**Comment by rnveach** [comment]
For https://github.com/checkstyle/checkstyle/pull/12707#discussion_r1094701977 ,

https://mvnrepository.com/artifact/org.pitest/pitest-maven
1.11.0 was released on 1/30. *(Missed, but being manually updated)*
1.9.10 was released on 11/11. *(Missed)*
1.9.9 was released on 10/25. *(Missed, but manually updated)*
1.9.8 was released on 10/10.

File: pom.xml (hunk 1 above)
Comment (rnveach) [reply pom.xml:213]: > [WARNING] Unable to exclude rules [BeanMembersShouldSerialize] from ruleset reference category/java/errorprone.xml; perhaps the rule name is misspelled or the rule doesn't exist anymore?
[WARNING] Discontinue using Rule name category/java/design.xml/ExcessiveMethodLength as it is scheduled for removal from PMD. PMD 7.0.0 will remove support for this Rule.

6.52
https://github.com/pmd/pmd/releases/tag/pmd_releases%2F6.52.0
> The Java rule [BeanMembersShouldSerialize](https://pmd.github.io/pmd-6.52.0/pmd_rules_java_errorprone.html#beanmembersshouldserialize) has been renamed to
[NonSerializableClass](https://pmd.github.io/pmd-6.52.0/pmd_rules_java_errorprone.html#nonserializableclass).

**Comment by rnveach** [comment]
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=13016&view=logs&j=1a6076f9-5826-5ba0-19ba-deaff2e35eea&t=5f46bc98-6f46-512e-6656-5fbfb1d03e4d&l=138 
Azure reported no new dependencies too on 1/29.

> cmd: ./.ci/validation.sh versions

https://github.com/checkstyle/checkstyle/blob/master/.ci/validation.sh#L172

It runs `versions:dependency-updates-report versions:plugin-updates-report` while I ran `versions:display-dependency-updates`.

File: pom.xml (hunk 1 above)
Comment (rnveach) [reply pom.xml:213]: https://pmd.github.io/latest/pmd_rules_java_design.html#immutablefield
Property ignoredAnnotations is listed as Deprecated.

**Comment by rnveach** [comment]
CI should pass now.

**Review by romani — APPROVED** [review]

**Pull request merged**